File: backend/src/main/java/com/elhabhab/backend/dto/request/OrderRequestDTO.java
package com.elhabhab.backend.dto.request;

import lombok.*;

import java.util.List;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class OrderRequestDTO {
    private UUID userId;
    private Long addressId;
    private List<OrderItemRequestDTO> items;
    private String couponCode; // optional
    private String deliveryOption; // "standard" or "express"
    private String paymentMethod; // "card" or "cod"


}

File: frontend-angular/src/app/interface/order-request-dto.ts
import { OrderItemRequest } from "./order-item-request";

export interface OrderRequestDTO {
  userId: string;
  addressId : number | null;
  couponCode?: string;
  deliveryOption: 'standard' | 'express'; // "standard" => 2-3 jours, "express" => 24h
  paymentMethod: 'cod' | 'card'; // méthode de paiement choisie
  items: OrderItemRequest[];
}